python/semantic_kernel/connectors/openapi/sk_openapi.py

```diff
@@ -0,0 +1,303 @@
+        self.logger = logger
+
+    """
+    Import an OpenAPI file.
+    :param openapi_file: The path to the OpenAPI file which can be local or a URL.
+    :return: The parsed OpenAPI file
+    """
+
+    def parse(self, openapi_document):
+        try:
+            parser = ResolvingParser(openapi_document)
+            return parser.specification
+        except ValidationError as e:
+            self.logger.debug(f"Error parsing OpenAPI document: {e}", exc_info=True)
+            raise e
+
+    """
+    Creates a RestApiOperation object for each path/method combination
+    :param parsed_document: The parsed OpenAPI document
+    :return: A dictionary of RestApiOperation objects keyed by operationId
+    """
+
+    def create_rest_api_operations(
+        self, parsed_document
+    ) -> Dict[str, RestApiOperation]:
+        paths = parsed_document.get("paths", {})
+        request_objects = {}
+        for path, methods in paths.items():
+            for method, details in methods.items():
+                server_url = parsed_document.get("servers", [])[0].get("url")
+
+                request_method = method.lower()
+
+                parameters = details.get("parameters", [])
+                operationId = details.get("operationId", path + "_" + request_method)
+                summary = details.get("summary", None)
+                description = details.get("description", None)
+
+                rest_api_operation = RestApiOperation(
+                    id=operationId,
+                    method=request_method,
+                    server_url=server_url,
+                    path=path,
+                    parameters=parameters,
+                    payload=details.get("requestBody", None),
+                    summary=summary,
+                    description=description,
+                )
+
+                request_objects[operationId] = rest_api_operation
+        return request_objects
+
+
+class OpenApiRunner:
+    def __init__(
+        self,
+        openapi_document: Dict[str, Union[str, List[str], Dict[str, str]]],
+        logger: logging.Logger = NullLogger(),
+    ):
+        self.logger = logger
+        self.spec = Spec.from_dict(openapi_document)
+
+    async def run_operation(
+        self,
+        operation: RestApiOperation,
+        path_params: Optional[Dict[str, str]] = None,
+        query_params: Optional[Dict[str, str]] = None,
+        headers: Optional[Dict[str, str]] = None,
+        payload: Optional[Union[str, Dict[str, str]]] = None,
+    ) -> aiohttp.ClientResponse:
+        prepared_request = operation.prepare_request(
+            path_params=path_params,
+            query_params=query_params,
+            headers=headers,
+            payload=payload,
+        )
+        is_valid = prepared_request.validate_request(spec=self.spec, logger=self.logger)
+        if not is_valid:
+            return None
+        try:
+            async with aiohttp.ClientSession(raise_for_status=True) as session:
```

Comment:
aiohttp wants you to reuse the ClientSession, but that will take some more work. It would be nice if the Kernel had a single session to use for all of these skills.